core/commands/block.go

```diff
@@ -282,38 +235,42 @@ It takes a list of base58 encoded multihashes to remove.
 		cmdkit.BoolOption("quiet", "q", "Write minimal output."),
 	},
 	Run: func(req *cmds.Request, res cmds.ResponseEmitter, env cmds.Environment) {
-		n, err := GetNode(env)
+		api, err := GetApi(env)
 		if err != nil {
 			res.SetError(err, cmdkit.ErrNormal)
 			return
 		}
-		hashes := req.Arguments
+
 		force, _ := req.Options["force"].(bool)
 		quiet, _ := req.Options["quiet"].(bool)
-		cids := make([]*cid.Cid, 0, len(hashes))
-		for _, hash := range hashes {
-			c, err := cid.Decode(hash)
+
+		// TODO: use batching coreapi when done
+		for _, b := range req.Arguments {
+			p, err := coreiface.ParsePath(b)
 			if err != nil {
-				err = fmt.Errorf("invalid content id: %s (%s)", hash, err)
 				res.SetError(err, cmdkit.ErrNormal)
 				return
 			}
 
-			cids = append(cids, c)
-		}
-		ch, err := util.RmBlocks(n.Blockstore, n.Pinning, cids, util.RmBlocksOpts{
-			Quiet: quiet,
-			Force: force,
-		})
+			rp, err := api.ResolvePath(req.Context, p)
+			if err != nil {
+				res.SetError(err, cmdkit.ErrNormal)
```

Comment:
Should we emit error here too?
What do you think?

Reply:
Reading original code we probably shouldn't unless `ResolvePath` does a lot more than just `Decode`.

Reply:
It does handle ipns and path traversal too